### ISSUE TITLE
In Modification class, getOldSourceCode

### ISSUE THREAD

**Issue by klerisson** [mauricioaniche/metricminer2#16]
What about adding `getOldSourceCode` in Modification class. I notice that `getSourceCode` give us the new one.

Let me know whether it would be interesting or not.

**Comment by mauricioaniche** [comment]
That's a good idea, yes.

However, some things for us to think:
- That would double the memory consumption. Source codes are long.
- I am not sure if the command MM2 uses to retrieve the current information from the commit (which retrieves the current source code), also retrieves the old one. If not, then that would be c*n more Git invokations, where c is the number of commits in the repo, and n the number of these files in commits.

Maybe a good alternative would be to provide a method in the SCM, such as getSourcePriorVersion(Modification, Commit), and then it extracts the information. Just like the 'checkout' API works now.

What do you think?

**Comment by klerisson** [comment]
I agree, even I'm also not sure about the second bullet. Anyway, it'll be able to provide the goods on demand.
